Modules/_threadmodule.c

```diff
@@ -1022,6 +1022,14 @@ rlock_traverse(PyObject *self, visitproc visit, void *arg)
     return 0;
 }
 
+/*
+helper function used by rlock_locked and rlock_repr.
+*/
+static int
+rlock_locked_impl(rlockobject *self) {
+    return PyMutex_IsLocked(&self->lock.mutex);
```

Comment:
Maybe use `return (self->lock.thread != 0);` and change the implementation in your second PR.